### ISSUE TITLE
PlaidLink is closed right away after the loading circle appears

### ISSUE THREAD

**Issue by jdconley** [plaid/react-native-plaid-link-sdk#620]
# The problem

The Link UI isn't displaying on our Android app. There are some warnings and errors in the logs that might be relevant.

# Environment

|                                                  |                                                                                  |
| -----------------------------| ----------------------------------------------- |
| Plaid Link React Native          | 11.2.0                                                                 |
| ReactNative Version               | 0.72.6                                                                 |
| Occurs on Android                  | yes                                                                     |
| Android OS Version                 | 14 (34)                                                         |
| Android Devices/Emulators    | Pixel 8 physical and various emulators                                                |
| Occurs on iOS                          | no                                                                     |
| iOS Version                               | 16+                                                                       |
| iOS Devices/Emulators            | all                                                    |
| Link Session ID                         | not sure, never see it in the logs |

# Steps to Reproduce

In order to get the app to compile I did some version mappings in gradle as shown in #608 :

```
    def lifecycle_version = "2.5.1"
    implementation "androidx.lifecycle:lifecycle-viewmodel:$lifecycle_version"
    implementation "androidx.lifecycle:lifecycle-viewmodel-ktx:$lifecycle_version"
    def work_version = "2.8.1"
    implementation "androidx.work:work-runtime:$work_version"
    implementation "androidx.work:work-runtime-ktx:$work_version"
```

Create a link token and then attempt to open the link UI. The UI doesn't display.

# Expected Result
Link UI to display.

# Screenshots

Link never displays it's just my app.

# Logs

```
2024-01-12 13:46:53.567 31753-31880 ReactNativeJS           com.myapp               I  Plaid link token created: link-production-5c2bc59c-4a8b-48f8-830d-bb2bd99977c6
2024-01-12 13:46:53.639 31753-31753 cr_CombinedPProvider    com.myapp               I  #registerProvider() provider:WV.v7@88a3583 isPolicyCacheEnabled:false policyProvidersSize:0
2024-01-12 13:46:53.641 31753-31753 cr_PolicyProvider       com.myapp               I  #setManagerAndSource() 0
2024-01-12 13:46:53.688 31753-31753 cr_CombinedPProvider    com.myapp               I  #linkNativeInternal() 1
2024-01-12 13:46:53.688 31753-31753 cr_AppResProvider       com.myapp               I  #getApplicationRestrictionsFromUserManager() Bundle[EMPTY_PARCEL]
2024-01-12 13:46:53.688 31753-31753 cr_PolicyProvider       com.myapp               I  #notifySettingsAvailable() 0
2024-01-12 13:46:53.688 31753-31753 cr_CombinedPProvider    com.myapp               I  #onSettingsAvailable() 0
2024-01-12 13:46:53.688 31753-31753 cr_CombinedPProvider    com.myapp               I  #flushPolicies()
2024-01-12 13:46:53.771 31753-32033 cr_TrackExitReasons     com.myapp               E  Failed to parse JSON from file.
2024-01-12 13:46:53.779 31753-31753 Compatibil...geReporter com.myapp               D  Compat change id reported: 214741472; UID 10275; state: ENABLED
2024-01-12 13:46:53.835 31753-31753 erhour.chat.dev         com.myapp               W  Accessing hidden method Ljava/lang/invoke/MethodHandles$Lookup;-><init>(Ljava/lang/Class;I)V (unsupported, reflection, allowed)
2024-01-12 13:46:53.864 31753-31753 erhour.chat.dev         com.myapp               W  Accessing hidden field Landroid/content/pm/ApplicationInfo;->primaryCpuAbi:Ljava/lang/String; (unsupported, reflection, allowed)
2024-01-12 13:46:53.884 31753-31753 WindowOnBackDispatcher  com.myapp               W  OnBackInvokedCallback is not enabled for the application.
                                                                                                    Set 'android:enableOnBackInvokedCallback="true"' in the application manifest.
2024-01-12 13:46:53.963 31753-32080 TrafficStats            com.myapp               D  tagSocket(146) with statsTag=0xffffffff, statsUid=-1
2024-01-12 13:46:54.754 31753-31753 WindowOnBackDispatcher  com.myapp               W  OnBackInvokedCallback is not enabled for the application.
                                                                                                    Set 'android:enableOnBackInvokedCallback="true"' in the application manifest.
2024-01-12 13:46:54.789 31753-32090 TrafficStats            com.myapp               D  tagSocket(305) with statsTag=0x90000, statsUid=-1
2024-01-12 13:46:54.873 31753-31753 Compatibil...geReporter com.myapp               D  Compat change id reported: 193247900; UID 10275; state: ENABLED
2024-01-12 13:46:54.889 31753-31753 libEGL                  com.myapp               D  loaded /vendor/lib64/egl/libGLES_mali.so
2024-01-12 13:46:54.958 31753-32036 cr_media                com.myapp               W  BLUETOOTH_CONNECT permission is missing.
2024-01-12 13:46:54.958 31753-32036 cr_media                com.myapp               W  registerBluetoothIntentsIfNeeded: Requires BLUETOOTH permission
2024-01-12 13:46:55.005 31753-31753 Compatibil...geReporter com.myapp               D  Compat change id reported: 78294732; UID 10275; state: ENABLED
2024-01-12 13:46:55.011 31753-31880 ReactNativeJS           com.myapp               I  Plaid link exit: {"metadata":{}}
2024-01-12 13:46:55.030 31753-32170 TrafficStats            com.myapp               D  tagSocket(197) with statsTag=0xffffffff, statsUid=-1
2024-01-12 13:46:55.067 31753-31753 WindowOnBackDispatcher  com.myapp               W  sendCancelIfRunning: isInProgress=falsecallback=android.view.ViewRootImpl$$ExternalSyntheticLambda17@f3dd28
2024-01-12 13:46:55.078 31753-31880 ReactNativeJS           com.myapp               I  [App State] Now: active
2024-01-12 13:46:55.084 31753-31753 chromium                com.myapp               E  [ERROR:aw_browser_terminator.cc(154)] Renderer process (32014) crash detected (code -1).
2024-01-12 13:46:55.114 31753-31812 Compatibil...geReporter com.myapp               D  Compat change id reported: 194532703; UID 10275; state: ENABLED
2024-01-12 13:46:55.116 31753-31812 Compatibil...geReporter com.myapp               D  Compat change id reported: 253665015; UID 10275; state: DISABLED
2024-01-12 13:46:56.236 31753-31880 ReactNativeJS           com.myapp               I  [AnalyticsService] Tracked "App Opened"
```

# Code To Reproduce Issue 

A very basic link open.

```
        openLink({
          tokenConfig: {
            token: linkTokenResponse.token,
            noLoadingState: true, // also tried false
            logLevel: LinkLogLevel.DEBUG,
          },
          onSuccess: success => this.onLinkSuccess(success),
          onExit: exit => this.onLinkExit(exit),
          iOSPresentationStyle: LinkIOSPresentationStyle.FULL_SCREEN,
        });
```

**Comment by jdconley** [comment]
I just created a brand new react native app (0.73.2) and tried to use the PlaidLink Pressable component with the same result. Though this time I didn't need to do the gradle hacks as above, so the plaid android sdk must line up better with the latest react native sdk.

**Comment by dtroupe-plaid** [comment]
@jdconley thanks for reporting. I am unable to reproduce this issue on Android 13. I used the token from your logs here are the results

![Screenshot_1705101839](https://github.com/plaid/react-native-plaid-link-sdk/assets/122908338/29b5bbd3-880a-4dc8-84c1-eda69df686c0)

https://github.com/plaid/react-native-plaid-link-sdk/assets/122908338/2c50baa9-3704-4e9d-a197-5072f9f5aefd

It also looks like there is a crash occurring in the logs 

```
[ERROR:aw_browser_terminator.cc(154)] Renderer process (32014) crash detected (code -1).
```

Does this issue only occur on Android 14?

**Comment by dtroupe-plaid** [comment]
@jdconley I've confirmed the issue is specific to Android 14 and I can reproduce it. I'll report back once I have an update.

**Comment by jdconley** [comment]
I also downgraded to the 10.x version of the Plaid Link RN SDK and had the same result. 9.x doesn't build with modern RN due to some gradle plugin incompatibilities.

**Comment by dtroupe-plaid** [comment]
@jdconley are you using the same token? 

I was able to reproduce the issue once, but after trying another token the UI did load as expected. Oddly enough it will also now load with the token provided in the logs 🤔

**Comment by jdconley** [comment]
> @jdconley are you using the same token? Do you have the same result on sandbox?
> 
> I was able to reproduce the issue once, but after trying another token the UI did load as expected. Oddly enough it will also now load with the token provided in the logs 🤔

That's... unfortunate. I was testing with that token but I can generate a new one. How about `link-production-b3b87048-f751-4ee1-9e8b-a8f68161adfa` -- just failed to launch link with it in my app. Haven't tried sandbox yet. Do you need my token creation arguments or can you look that up internally?

**Comment by dtroupe-plaid** [comment]
It's very inconsistent for me. Some times it loads and some times it exits right away. I think I have all the information I need for now. Thanks for the help 🙏

**Comment by ChanukaSandeepa** [comment]
Still UI does not load even with newly created tokens on SandBox. The UI loads with an error message only when the token is expired. Do you have some suggestions?

**Comment by jdconley** [comment]
I just loaded up Android 13 in a simulator to see if I could test the feature there. It is reproducing for me on Android 13 API 33 as well in the arm64 simulator.

**Comment by dtroupe-plaid** [comment]
@ChanukaSandeepa @jdconley I was able to reproduce this issue in [version 4.1.0](https://github.com/plaid/plaid-link-android/releases/tag/v4.1.0) of our android SDK. 

We are working to resolve the issue. In the meantime you can use a version of the RN sdk that doesn't use version `4.1.0` or `3.14.2` that means you want to use version [v11.1.0](https://github.com/plaid/react-native-plaid-link-sdk/releases/tag/v11.1.0) or [v10.9.0](https://github.com/plaid/react-native-plaid-link-sdk/releases/tag/v10.9.0). 

Please let me know if you have issues with either of those SDKs. I will post an timeline update as soon as I have one. Thanks for your patience 🙏

**Comment by jdconley** [comment]
> We are working to resolve the issue. In the meantime you can use a version of the RN sdk that doesn't use version `4.1.0` or `3.14.2` that means you want to use version [v11.1.0](https://github.com/plaid/react-native-plaid-link-sdk/releases/tag/v11.1.0) or [v10.9.0](https://github.com/plaid/react-native-plaid-link-sdk/releases/tag/v10.9.0).

Confirmed the Link UI loads with v11.1.0 for me.

**Comment by ChanukaSandeepa** [comment]
The UI does not loads for v11.1.0 and I'm getting a crash for v10.9.0. The crash says `PlaidAndroid.startLinkActivityForResult got 3 arguments, expected 5`. Do you have any suggestions for these 5 arguments?

**Comment by dtroupe-plaid** [comment]
Hi

I've released [v11.3.0](https://github.com/plaid/react-native-plaid-link-sdk/releases/tag/v11.3.0) && [v10.11.0](https://github.com/plaid/react-native-plaid-link-sdk/releases/tag/v10.11.0) which revert the Android SDK to 4.0.0 which doesn't have this issue. 

Additionally, versions 11.2.0 and 10.10.0 are deprecated. 

I will leave this issue open until we have an updated Android SDK and we confirm the issue no longer exists. For now using the latest SDK will avoid this problem 🙏

**Comment by dtroupe-plaid** [comment]
Releases [11.4.0](https://github.com/plaid/react-native-plaid-link-sdk/releases/tag/v11.4.0) and [v10.12.0](https://github.com/plaid/react-native-plaid-link-sdk/releases/tag/v10.12.0) were released today to address this issue. I am going to close this issue.
